hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -434,8 +435,12 @@ ContainerCommandResponseProto handleCreateContainer(
         LOG.debug("Container already exists. container Id {}", containerID);
       }
     } catch (StorageContainerException ex) {
+      exceptionThrown = true;
       return ContainerUtils.logAndReturnError(LOG, ex, request);
     } finally {
+      if (exceptionThrown) {
```

Comment:
1.  please move this newContainerData.releaseCommitSpace() into the exception handling directly. 
2. we might first check whether space is committed before release, for newContainer.create can throw out exception from volumeChoosingPolicy.chooseVolume  too.

Reply:
Okay.